### ISSUE TITLE
AODA issue caused by Issue #2771

### ISSUE THREAD

**Issue by tmdaly** [Leaflet/Leaflet#6434]
The addition of the extra DIV that prevents the layer control flicker when checkboxes are disabled causes [AODA ](https://www.aoda.ca/) issues because it's not semantic HTML.

**Steps to reproduce**
- Inspect the page with dev tools
- From the Elements tab of the dev tools, right click on the <html> element and select Copy > Copy element
- Navigate to https://validator.w3.org/nu/#textarea
- Retain the doctype in the sample HTML in the text input; replace the rest of the sample HTML with your copied HTML from your page
- Click Check

**Expected behavior**
The expected behavior is that the page produces no errors

**Current behavior**
There are 6 errors related to the extra DIV placed in a LABEL as a fix for Issue #2771

**Environment**
- Leaflet version: 1.3.1
- Browser (with version): Chrome
- OS/Platform (with version): Windows 10

**Comment by Falke-Design** [comment]
It looks like this is resolved. I can't reproduce the errors with the [layer control sample](https://leafletjs.com/examples/layers-control/example.html)

@Malvoz what do you think?

**Comment by Malvoz** [comment]
@Falke-Design this issue is about HTML validation errors, I see only 2 related to the layer control:

> **Error**: Element `div` not allowed as child of element `label` in this context. (Suppressing further errors from this subtree.)
> `<label><div><input`

I don't think it's likely this'll cause any accessibility issues, OTOH valid HTML is probably desired so perhaps it'd be worth to investigate if the extra `<div>` inside `<label>` can now be removed without causing issues such as https://github.com/Leaflet/Leaflet/issues/2771, or replace these `div` elements with `span`s since that's valid HTML.

> **Warning**: Section lacks heading. Consider using `h2`-`h6` elements to add identifying headings to all sections.
> `<section class="leaflet-control-layers-list">`

This should be fixed by https://github.com/Leaflet/Leaflet/issues/7530.
